File: C++/2597. The Number of Beautiful Subsets.cpp
class Solution {
public:
    int beautifulSubsets(vector<int>& nums, int k) {
        std::sort(nums.begin(), nums.end());
        int answer = 0;
        int n = (int) nums.size();
        std::vector<int> f;
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                if (nums[j] - nums[i] == k) {
                    f.push_back((1 << i) | (1 << j));
                }
            }
        }
        for (int mask = 1; mask < (1 << n); ++mask) {
            bool ok = true;
            for (int v : f) {
                if ((mask & v)== v) {
                    ok = false;
                    break;
                }
            }
            answer += ok;
        }
        return answer;
    }
};

File: JAVA/2597. The Number of Beautiful Subsets.java
class Solution {
    ArrayList<ArrayList<Integer>> arr = new ArrayList<ArrayList<Integer>>();
    public void find(int ind,ArrayList<Integer> temp,int[] nums,int k)
    {
        if(ind==nums.length)
        {
            arr.add(temp);
            return;
        }
        if(temp.indexOf(nums[ind]-k)<0)
        {
            temp.add(nums[ind]);
            find(ind+1,temp,nums,k);
            temp.remove(temp.size()-1);
        }
        find(ind+1,temp,nums,k);
    }
    public int beautifulSubsets(int[] nums, int k) {
        if(nums.length==0 || nums.length==1)
            return nums.length;
        Arrays.sort(nums);
        ArrayList<Integer> temp = new ArrayList<Integer>();
        find(0,temp,nums,k);
        return arr.size()-1;
    }   
}
